static/js/show-task.js

```diff
@@ -39,20 +40,39 @@ function saveTask() {
     }});
 }
 
+function rememberState() {
+        // Finding how many rows are being displayed so that we can make it same after refresh
+        var rowsPerPageDropdownIndex = $('.page-list .dropdown-menu .dropdown-item.active').index();
+
+        // Finding which page is active so that we can make it active again after refresh
+        rowsPerPageIndex = $('.pagination .page-item.active').index();
+
+        return {'rowsPerPageDropdownIndex': rowsPerPageDropdownIndex,
+                'rowsPerPageIndex': rowsPerPageIndex
+        }
+}
+
+function restoreState() {
+      // making the number of rows to be same
+      var dropdownElements = $('.page-list .dropdown-menu .dropdown-item');
+      dropdownElements[rowsPerPageIndices.rowsPerPageDropdownIndex].click();
+
+      // making the same page active as was before
+      var paginationElements = $('.pagination .page-item .page-link');
+      paginationElements[rowsPerPageIndices.rowsPerPageIndex].click();
+}
+
+
 function refresh() {
     search_term = $("#search_by_task_name_form input[name=search_term]").val();
     worklist_name = $("#search_by_task_name_form input[name=worklist_name]").val();
     worklist_created_by = $("#search_by_task_name_form input[name=worklist_created_by]").val();
 
-    $('#task_table').html('');
-    $('#task_table').addClass('loader');
-    $('#task_table').html('').load(
-    "/worklist-tool/update-task-table?search_term=" + encodeURIComponent(search_term) + "&worklist_name=" +
-        encodeURIComponent(worklist_name) + "&worklist_created_by=" + encodeURIComponent(worklist_created_by),
-    function() {
-      $('#task_table').removeClass('loader');
-    }
-    );
+    rowsPerPageIndices = rememberState();
+
+    $('#task_table').load(
+        "/worklist-tool/update-task-table/" + encodeURIComponent(worklist_created_by) + "/" + encodeURIComponent(worklist_name) + "?search_term=" + encodeURIComponent(search_term)
+    ).hide().fadeIn('slow');
```

Comment:
I'm seeing the fade-in effect when we begin the background request, and then the table actually updates a few seconds later as the request finishes. Was this intended?

If we're going to have this hide/fade-in effect on the whole table, I think we probably want the following:

1- Start the background request
2- When it completes, hide the table, update its innerHTML.
3- Fade back in.

So:

```js
$.get(..., function(data) {
  var $table = $('#task_table');
  $table.hide();
  $table.html(data);
  $table.fadeIn('slow');
});
```

Depending on how `$.html` works we might need to put the `fadeIn` call in a `setTimeout(..., 0)` to possibly [allow the browser to apply the new HTML](https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful). How does this sound?

Reply:
That wasn't intended really, and it wasn't happening with me. But still, I've improved the implementation by doing all of this in success callback. Can you please confirm if it looks fine now? If it does, shall I merge this PR? I've another one ready.

Reply:
> That wasn't intended really, and it wasn't happening with me. 

It might depend on the worklist you test with -- I usually go for my Colombia one, which is large so the refresh request takes multiple seconds. For a small worklist, I would imagine the request would be so fast that the bug would be hard to notice.